### PR TITLE
[Doc]Update flex flow doc 

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.
This pull request includes changes to the `docs/how-to-guides/develop-a-flex-flow` directory, specifically the `class-based-flow.md` and `function-based-flow.md` files. The changes primarily involve the addition of new configurations and parameters to the examples provided in these files. 

* [`docs/how-to-guides/develop-a-flex-flow/class-based-flow.md`](diffhunk://#diff-0aa42d1976c0f4db59cb05a81413c6dafc5f6c6bb6f9c82961299aad956fd2e6R97-R108): Added `AzureOpenAIModelConfiguration` to the example and updated the `pf.run` method to include the `model_config` and `flow_config` parameters. This provides a more detailed example of how to run a flow class directly with specific configurations.
* [`docs/how-to-guides/develop-a-flex-flow/function-based-flow.md`](diffhunk://#diff-af24f978caaac273a3a303a2271b7bd374c55c87bf6c1bca283b796df3efd31bR96): Added `inputs` to the `sample` section of the flow YAML example. This shows how to specify inputs in a flow YAML file.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by brynn-code — APPROVED** [review]

**Review by Stephen1993 — APPROVED** [review]

File: docs/how-to-guides/develop-a-flex-flow/function-based-flow.md [hunk 1]
```diff
@@ -93,7 +93,8 @@ A flow YAML may look like this:
 $schema: https://azuremlschemas.azureedge.net/promptflow/latest/Flow.schema.json
 entry: path.to.module:function_name
 sample:
-  question: "what's the capital of France?"
```
Comment (Stephen1993) [inline docs/how-to-guides/develop-a-flex-flow/function-based-flow.md:96]: I have a concern, will any users use it this way before? And our latest version of promptflow doesn't support this?

Comment (D-W-) [reply docs/how-to-guides/develop-a-flex-flow/function-based-flow.md:96]: Good catch. Seems other docs' sample field doesn't have inputs. Let me remove it to reduce confusion

Comment (D-W-) [reply docs/how-to-guides/develop-a-flex-flow/function-based-flow.md:96]: Synced with @wangchao1230 , according to current implementation, flows will only support sample with inputs. Prompty will support sample without inputs.

File: docs/how-to-guides/develop-a-flex-flow/connection-support.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Connection support
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-a-flex-flow/connection-support.md:1]: better title

File: docs/how-to-guides/develop-a-flex-flow/connection-support.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## Model config in `__init__`
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-a-flex-flow/connection-support.md:3]: introduce what's a model config first

File: docs/how-to-guides/develop-a-flex-flow/connection-support.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+```python
+class MyFlow:
+    def __init__(self, my_connection: AzureOpenAIConnection):
+        pass
+```
+
+Note:
+
+- Union of connection types(`Union[OpenAIConnection, AzureOpenAIConnection]`) is not supported.
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-a-flex-flow/connection-support.md:22]: do we have some page introduce the model config?

File: docs/how-to-guides/develop-a-flex-flow/class-based-flow.md [hunk 5]
```diff
@@ -37,6 +37,8 @@ class MyFlow:
 
 ## Flow test
 
+### Test with original code
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-a-flex-flow/class-based-flow.md:40]: "original code" is hard to understand for first-use user
Not sure if below is better?
Test via function call

File: docs/how-to-guides/develop-a-flex-flow/connection-support.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+In cloud, the connection name will be replaced with workspace's connection object in execution time.
+
+```python
+# local connection "my_connection"'s instance will be passed to `__init__`
+pf.run(flow="./flow.flex.yaml", init={"connection": "my_connection"}, data="./data.jsonl")
+# cloud connection "my_cloud_connection"'s instance will be passed to `__init__`
+pfazure.run(flow="./flow.flex.yaml", init={"connection": "my_cloud_connection"}, data="./data.jsonl")
+```
+
+## Environment variable connections(EVC)
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-a-flex-flow/connection-support.md:38]: no need to mention EVC if it's not used below

File: docs/how-to-guides/develop-a-flex-flow/connection-support.md (hunk 3 above)
Comment (D-W-) [reply docs/how-to-guides/develop-a-flex-flow/connection-support.md:6]: Added a page for model config

File: docs/how-to-guides/develop-a-flex-flow/connection-support.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+class MyFlow:
+    def __init__(self, my_connection: AzureOpenAIConnection):
+        pass
+```
+
+Note:
+
+- Union of connection types(`Union[OpenAIConnection, AzureOpenAIConnection]`) is not supported.
+
+### Batch run with connection
```
[the remark this answers is not in the document]
Comment (D-W-) [reply docs/how-to-guides/develop-a-flex-flow/connection-support.md:28]: Added link under title

File: docs/how-to-guides/develop-a-flex-flow/connection-support.md (hunk 4 above)
Comment (D-W-) [reply docs/how-to-guides/develop-a-flex-flow/connection-support.md:26]: Added a page for it

File: docs/how-to-guides/develop-a-flex-flow/model-config.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# Model Configuration
+
+Tailoring your AI applications to use specific models with desired parameters is now easier than ever with the Promptflow SDK's new Model Configuration feature. Whether you're engaging with chatbots, generating text, or building complex AI workflows, our unified configuration approach allows you to set up and switch between models from OpenAI, AzureOpenAI, MistralAI, and other LLM endpoints seamlessly.
```
[the remark this answers is not in the document]
Comment (wangchao1230) [reply docs/how-to-guides/develop-a-flex-flow/model-config.md:3]: Update here, as MistralAI and other LLM endpoints support not available yet.

[the remark this answers is not in the document]
Comment (D-W-) [reply docs/how-to-guides/develop-a-flex-flow/model-config.md:3]: Chat GPT generated from model config's spec

**Review by elliotzh — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**